### ISSUE TITLE
Rule proposal: `no-unnecessary-await`

### ISSUE THREAD

**Issue by fisker** [sindresorhus/eslint-plugin-unicorn#1558]
This prevents mistakes.

## Fail

```js
const foo = await await bar;
```

## Pass

```js
const foo = await bar;
```

---

Or maybe a more generic rule forbid `await` on non-promise objects?
So we can report `await 1`, `await true`, `await nonPromiseObject` etc.
This rule will cover the case above, because `await` result can't be a promise.

**Comment by sindresorhus** [comment]
This is accepted.

**Comment by fisker** [comment]
The rule name `await-thenable` from `typescript-eslint` seems a little confusing, it's like "the thenable should be `await`ed", how about `no-await-non-thenable`?

**Comment by papb** [comment]
@fisker Maybe `no-invalid-await` or `no-non-thenable-await`?

**Comment by fisker** [comment]
`no-invalid-await` is not good, since it's actually valid, just unnecessary.

**Comment by papb** [comment]
@fisker Oops you're right. So `no-unnecessary-await`?
